lib/core/job_cleanup.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core'
+require 'utility/logger'
+
+module Core
+  class JobCleanUp
+    class << self
+
+      def execute(connector_id = nil)
+        # clean up orphaned jobs
+        Utility::Logger.info("Start cleaning up orphaned jobs for #{connector_id ? "connector #{connector_id}" : 'native connectors'}...")
+        orphaned_jobs = connector_id ? orphaned_jobs_for_single_connector(connector_id) : orphaned_jobs_for_native_connectors
```

Comment:
if a non-native connector is only cleaning up orphaned jobs that it is directly associated with, does that not mean that an operator needs to:
1. delete a connector
2. wait a bit
3. stop their connector instance

in that particular order? If they instead:

1. stop their connector instance
2. delete the index/connector

the orphaned job record will never be cleaned.

Reply:
That is true.

Ideally, when an operator deletes a connector in Kibana, all the associated jobs should be deleted. Is this the case in 8.6? cc @sphilipse 

I explicitly stated in the implementation doc, that a connector service should only clean up jobs, the service types of which are supported. But thinking twice, we should probably just delete all orphaned jobs, regardless of their service types. It does no harm in both product-level (no orphaned jobs will be left as long as there's at least one connector service running) and engineering-level (we use `terms` query and `delete_by_query` so we don't need to worry about race condition) cc @danajuratoni @tarekziade

Reply:
We probably need to revisit this logic ASAP when cancellation is available and not allow to delete indices that have ongoing work, that'll make this code redundant and logic a bit simpler.

If people would want to do manual deletion of jobs/indices, they can break it, but that's okay IMO